### PR TITLE
Generate token lists for nova 

### PR DESCRIPTION
**Pull request by shbhakta**

### CODE REVIEW

File: src/lib/utils.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import path from 'path'
 import yargs from "./getClargs"
 
 export const isNova = yargs.l2NetworkID === 42170;
+export const isArb1 = yargs.l2NetworkID === 42161;
```
Comment (shbhakta) [inline src/lib/utils.ts:16]: do we need isArb1 anywhere if we're checking for isNova?

File: src/lib/token_list_gen.ts [hunk 2]
```diff
@@ -76,106 +127,203 @@ export const generateTokenList = async (
...
+        ),
+        (error) => getTokens(
+          l1TokenList.tokens.map((token) => ({
+            addr: token.address.toLowerCase(),
+            logo: token.logoURI
+          })),
+          l2.network.chainID
+        ), 
+        3, 
+        false);
```
Comment (fredlacs) [inline src/lib/token_list_gen.ts:151]: there is a default value of false for this, so we don't need to add to every time we call

File: src/lib/token_list_gen.ts [hunk 3]
```diff
@@ -76,106 +127,203 @@ export const generateTokenList = async (
...
-
+        ),
+        (error) => getTokens(
+          l1TokenList.tokens.map((token) => ({
+            addr: token.address.toLowerCase(),
+            logo: token.logoURI
+          })),
+          l2.network.chainID
+        ), 
+        3,
```
Comment (fredlacs) [inline src/lib/token_list_gen.ts:150]: seems like we use this value everywhere, lets add it as a default value to simplify calling the `promiseErrorMultiplier`

File: src/lib/token_list_gen.ts [hunk 4]
```diff
@@ -76,106 +127,203 @@ export const generateTokenList = async (
...
-  const l2AddressesFromL2 = await getL2TokenAddressesFromL2(l1TokenAddresses, l2.multiCaller, l2.network.tokenBridge.l2GatewayRouter);
+  if(isNova) {
+    const logos = l1TokenList.tokens.reduce(
+      (acc, curr) => ((acc[curr.address.toLowerCase()] = curr.logoURI), acc),
+      {} as { [addr: string]: string | undefined }
+    );
+    
+    let intermediatel2GatewayAddrs = [];
+    for (const addrs of getChunks(l1TokenAddresses)){
+      let l2GatewayAddressesFromL1Temp = await promiseErrorMultiplier(getL2GatewayAddressesFromL1Token(addrs, l2.multiCaller, l2.network), (error) => getL2GatewayAddressesFromL1Token(addrs, l2.multiCaller, l2.network), 3, false);
```
Comment (fredlacs) [inline src/lib/token_list_gen.ts:177]: would be good to break this down into multiple lines (or use prettier formatter to lint it)

**Review by fredlacs — CHANGES_REQUESTED** [review]

**Review by fredlacs — APPROVED** [review]
LGTM

**Pull request merged**